coa/pkg/apis/v1alpha2/providers/states/redisstate/redisstate.go

```diff
@@ -147,9 +177,17 @@ func (r *RedisStateProvider) Upsert(ctx context.Context, entry states.UpsertRequ
 	if err != nil {
 		return entry.Value.ID, err
 	}
+	firstWrite := 0
+	if entry.Options.Concurrency == states.FirstWrite {
```

Comment:
Does this mean "first write win"?

Reply:
Yes. If two writes with the same ETag coming in, the first write will win and bump etag while the second write fails.

Reply:
Hi @RemindD , I think this PR will be a little overlapped with #568, let's finalize #568 at first then go through this PR.